src/richie/apps/search/indexers/courses.py

```diff
@@ -603,6 +606,21 @@ def get_es_document_for_course(cls, course, index=None, action="index"):
             .values("start", "end", "enrollment_start", "enrollment_end", "languages")
         ]
 
+        try:
+            licences = [
+                licence.licence.id
+                for licence in LicencePluginModel.objects.filter(
+                    id__in=[
+                        plugin.id
+                        for plugin in course.extended_object.get_placeholders()
+                        .get(slot="course_license_content")
+                        .get_plugins()
+                    ]
+                )
+            ]
+        except (LicencePluginModel.DoesNotExist, Placeholder.DoesNotExist):
+            licences = None
```

Comment:
I proposed a fixup with an alternative after writing a test in `test_indexers_courses` which surfaced a duplicate result issue. There was actually a way to make the query directly on the Licence :sweat_smile: